File: AmtOrk.Server/Models/kingdom.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.RegularExpressions;

namespace AmtOrk.Server.Models
{
	public class Kingdom : RootModel
	{
		public string KingdomName { get; set; } = default!;
		public string? KingdomDesciption { get; set;}
		public string? KingdomIconPath { get; set; }
		public virtual List<Land> Lands { get; set; } = default!;
		public virtual AmtUser? Monarch { get; set; }
		public virtual AmtUser? Regent { get; set; }
		public virtual AmtUser? Champion { get; set; }
		public virtual AmtUser? PrimeMinister { get; set; }
		public virtual AmtUser? GMR { get; set; }
		public virtual List<Event>? KingdomEvents { get; set; }
		[NotMapped]
		public virtual string KingdomSlug { get => Regex.Replace(KingdomName.ToLower(), @"\s", "-");}
	}
}

File: amtork.client/src/app/models/kingdom.ts
import { AmtgardEvent } from "./amtgardEvent";
import { AmtUser } from "./amtUser";
import { Land } from "./land";

export interface Kingdom
{
	id:number;
	kingdomName:string;
	lands:Land[];
	monarch:AmtUser | undefined;
	regent:AmtUser | undefined;
	champion:AmtUser | undefined;
	primeMinister:AmtUser | undefined;
	gmr:AmtUser | undefined;
	kingdomEvents:AmtgardEvent[];
	readonly kingdomSlug:string;
}

export const emptyKingdom:Kingdom =
{
	id: 0,
	kingdomName: "",
	lands: [],
	monarch: undefined,
	regent: undefined,
	champion: undefined,
	primeMinister: undefined,
	gmr: undefined,
	kingdomEvents: [],
	get kingdomSlug():string {
		return this.kingdomName.toLowerCase().replace(" ","-");
	}
}